includes/discovery/discovery-protocols.inc.php

```diff
@@ -272,39 +279,43 @@ function ($v) {
 
                 $remote_device = device_by_id_cache($remote_device_id);
                 if ($remote_device['os'] == 'calix') {
-                    $lldp['lldpRemPortId'] = 'EthPort ' . $lldp['lldpRemPortId'];
+                    $remote_port_name = 'EthPort ' . $lldp['lldpRemPortId'];
                 }
 
                 if ($remote_device['os'] == 'xos') {
-                    $slot_port = explode(':', $lldp['lldpRemPortId']);
+                    $slot_port = explode(':', $remote_port_name);
                     if (sizeof($slot_port) == 2) {
                         $n_slot = (int) $slot_port[0];
                         $n_port = (int) $slot_port[1];
                     } else {
                         $n_slot = 1;
                         $n_port = (int) $slot_port[0];
                     }
-                    $lldp['lldpRemPortId'] = (string) ($n_slot * 1000 + $n_port);
+                    $remote_port_name = (string) ($n_slot * 1000 + $n_port);
                 }
 
                 $remote_port_id = find_port_id(
                     $lldp['lldpRemPortDesc'],
-                    $lldp['lldpRemPortId'],
+                    $remote_port_name,
                     $remote_device_id,
                     $remote_port_mac
                 );
-
+                if ($remote_port_id == 0) { //We did not find it
+                    $remote_port_name = $remote_port_name . ' (' . $remote_port_mac . ')';
+                }
                 if (empty($lldp['lldpRemSysName'])) {
                     $lldp['lldpRemSysName'] = $remote_device['sysName'] ?: $remote_device['hostname'];
                 }
-
-                if ($interface['port_id'] && $lldp['lldpRemSysName'] && $lldp['lldpRemPortId']) {
+                if (empty($lldp['lldpRemSysName'])) {
+                    $lldp['lldpRemSysName'] = $lldp['lldpRemSysDesc'];
+                }
```

Comment:
Some devices do not provide a SysName. They were skipped. This patch allows to use sysDesc in that case, and allow discovery. I had EMULEX cards in servers in that case (the card is directly sending LLDP packets independently of the OS).